### PR TITLE
Add `status` and `last_polled` to deployment API responses

### PR DESCRIPTION
**Pull request by serinamarie**
This PR allows deployment status to be determined at query time from `deployment.last_polled` time or `deployment.work_queue.last_polled` time. If it was polled in the last 30 seconds, `deployment.status` is `READY` else `NOT_READY`.
### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: src/prefect/server/schemas/responses.py [hunk 1]
```diff
@@ -232,6 +232,9 @@ def __eq__(self, other: Any) -> bool:
         return super().__eq__(other)
 
 
+DEPLOYMENT_LAST_POLLED_EXPIRATION = 30
```
Comment (serinamarie) [inline src/prefect/server/schemas/responses.py:235]: not sure on this name

File: src/prefect/server/schemas/responses.py [hunk 2]
```diff
@@ -261,13 +265,33 @@ class DeploymentResponse(ORMBaseModel):
...
 
     @classmethod
     def from_orm(
         cls, orm_deployment: "prefect.server.database.orm_models.ORMDeployment"
     ):
         response = super().from_orm(orm_deployment)
+
+        offline_horizon = datetime.datetime.now(
+            tz=datetime.timezone.utc
+        ) - datetime.timedelta(seconds=DEPLOYMENT_LAST_POLLED_TIMEOUT_SECONDS)
```
Comment (serinamarie) [inline src/prefect/server/schemas/responses.py:282]: happy to rename this to something like `not_ready_horizon` if folks don't think this is aptly named

File: src/prefect/client/schemas/objects.py [hunk 3]
```diff
@@ -956,6 +963,10 @@ class Deployment(ObjectBaseModel):
             " be scheduled."
         ),
     )
+    last_polled: Optional[DateTimeTZ] = Field(
+        default=None,
+        description="The last time the deployment was polled for status updates.",
+    )
```
Comment (serinamarie) [inline src/prefect/client/schemas/objects.py:969]: I'm currently just including `status` in the special response schemas

File: src/prefect/server/schemas/responses.py [hunk 4]
```diff
@@ -261,18 +265,36 @@ class DeploymentResponse(ORMBaseModel):
...
+
+        if response.last_polled and response.last_polled > offline_horizon:
+            response.status = schemas.statuses.DeploymentStatus.READY
+        elif (
+            orm_deployment.work_queue
+            and orm_deployment.work_queue.last_polled
+            and orm_deployment.work_queue.last_polled > offline_horizon
+        ):
+            response.status = schemas.statuses.DeploymentStatus.READY
```
Comment (serinamarie) [inline src/prefect/server/schemas/responses.py:296]: I felt it was more readable to separate these conditions but welcome other thoughts

File: src/prefect/server/schemas/responses.py (hunk 2 above)
Comment (desertaxle) [reply src/prefect/server/schemas/responses.py:287]: Yeah, I prefer `not_ready_horizon` since it uses the established nomenclature for deployment status.

Comment (serinamarie) [reply src/prefect/server/schemas/responses.py]: Renamed in [a0b989b](https://github.com/PrefectHQ/prefect/pull/10951/commits/a0b989beb477762efdf76906b82296b75dfca8d6)

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**